cpp/examples/arrow/row_wise_conversion_example.cc

```diff
@@ -47,10 +51,10 @@ struct data_row {
 // construction of the final `arrow::Array` instances.
 //
 // For each type, Arrow has a specially typed builder class. For the primitive
-// values `id` and `cost` we can use the respective `arrow::Int64Builder` and
-// `arrow::DoubleBuilder`. For the `cost_components` vector, we need to have two
-// builders, a top-level `arrow::ListBuilder` that builds the array of offsets and
-// a nested `arrow::DoubleBuilder` that constructs the underlying values array that
+// values `id` and `components` we can use the `arrow::Int64Builder`. For the 
+// `component_cost` vector, we need to have two builders, a top-level 
+// `arrow::ListBuilder` that builds the array of offsets and a nested 
+// `arrow::DoubleBuilder` that constructs the underlying values array that
 // is referenced by the offsets in the former array.
 arrow::Status VectorToColumnarTable(const std::vector<struct data_row>& rows,
                                     std::shared_ptr<arrow::Table>* table) {
```

Comment:
Since we are updating this function, can it perhaps return `Result<std::shared_ptr<Table>>` instead of `Status`?

See https://arrow.apache.org/docs/cpp/conventions.html#error-reporting

Reply:
A minimal example of the required conversion is at https://github.com/bkmgit/arrow/blob/bkmgit-error-reporting-example/cpp/examples/arrow/error_report_result.cc

Reply:
@pitrou Done